x-pack/plugins/security_solution/common/endpoint/models/event.ts

```diff
@@ -52,3 +52,123 @@ export function parentEntityId(event: ResolverEvent): string | undefined {
+      eventCategoryToReturn = category;
+    }
+  }
+  return eventCategoryToReturn;
+}
+
+/**
+ * ECS event type will be things like 'creation', 'deletion', 'access', etc.
+ * see: https://www.elastic.co/guide/en/ecs/current/ecs-event.html
+ * @param event The ResolverEvent to get the ecs type for
+ */
+export function ecsEventType(event: ResolverEvent): string {
+  if (isLegacyEvent(event)) {
+    return event.endgame.event_subtype_full || '';
+  }
+  return typeof event.event.type === 'string' ? event.event.type : event.event.type.join('/');
+}
+
+/**
+ * #Descriptive Names For Related Events:
+ *
+ * The following section provides facilities for deriving **Descriptive Names** for ECS-compliant event data.
+ * There are drawbacks to trying to do this: It *will* require ongoing maintenance. It presents temptations to overarticulate.
+ * On balance, however, it seems that the benefit of giving the user some form of information they can recognize & scan outweighs the drawbacks.
+ */
+
+/**
+ * A type that annotates the basic requirements for giving the event a `descriptive name`
+ */
+type detailRecord<T> = T extends 'registry'
+  ? { path: unknown; key: unknown }
+  : T extends 'dns'
+  ? { question: { name: unknown } }
+  : T extends 'network'
+  ? { direction: unknown; forwarded_ip: unknown }
+  : T extends 'file'
+  ? { path: unknown }
+  : unknown;
+type ecsRecordWithType<T extends string> = Record<T, detailRecord<T>>;
+
+/**
+ * Verify that the `ecsCategory` exists as a key on the event record. i.e. if ecsCategory is `registry`, the event is shaped like {registry: object}
+ *
+ * @param event
+ * @param ecsCategory
+ */
+export function isRecordNamable<T extends string>(
+  event: object,
+  ecsCategory: T
+): event is ecsRecordWithType<T> {
+  // The goal here is to reach in and grab a better name for the event (if one exists) without being too obtrusive/assertive about ECS
+  return typeof (event as ecsRecordWithType<T>)[ecsCategory] === 'object';
+}
+
+/**
+ * Based on the ECS category of the event, attempt to provide a more descriptive name
+ * (e.g. the `event.registry.key` for `registry` or the `dns.question.name` for `dns`, etc.).
+ * see: https://www.elastic.co/guide/en/ecs/current/ecs-field-reference.html
+ * @param event The ResolverEvent to get the descriptive name for
+ * @returns { descriptiveName } An attempt at providing a readable name to the user
+ */
+export function descriptiveName(event: ResolverEvent): string {
+  if (isLegacyEvent(event)) {
+    return eventName(event);
+  }
+  // For the purposes of providing a descriptive name, we're taking the first entry in the `event.type`
+  const ecsCategory = eventCategory(event);
+
+  /**
+   * This list of attempts can be expanded/adjusted as the underlying model changes over time:
+   */
+
+  // Stable, per ECS 1.5: https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html
+  if (ecsCategory === 'network' && isRecordNamable(event, ecsCategory)) {
```

Comment:
Could be more readable to document the boolean checks like this? 
```
const isNetwork = ecsCategory === 'network' && isRecordNamable(event, ecsCategory) && event?.network?.forwarded_ip;

const isFile = .....

const isRegistry = .....
.....


if (isNetwork) {
	return `${event?.network?.direction ? `${event?.network?.direction} ` : ''}${
        event.network.forwarded_ip
      }`;
}

...
```

Reply:
That looks better, I'm gonna try that. Does it still look good if I call it `isNetworkNamable` `isFileNamable` ?

Reply:
With your advice and a nudge froim Typescript, ended up with this: https://github.com/elastic/kibana/pull/69098/commits/be839a21ab45d854f09789b8eb18124ef57687fd